task_processing/plugins/mesos/retrying_executor.py

```diff
@@ -0,0 +1,60 @@
+import time
+from threading import Thread
+
+from pyrsistent import m
+from six.moves.queue import Queue
+
+from task_processing.interfaces.task_executor import TaskExecutor
+
+
+class RetryingExecutor(TaskExecutor):
+    def __init__(self,
+                 executor,
+                 retry_pred=lambda e: not e.success,
+                 retries=3):
+        self.executor = executor
+        self.retries = retries
+        self.retry_pred = retry_pred
+        self.task_retries = m()
+        self.src_queue = executor.get_event_queue()
+        self.dest_queue = Queue()
+        self.retry_thread = Thread(target=self.retry_loop)
+        self.retry_thread.start()
+        self.stopping = False
+
+    def retry_loop(self):
+        while True:
+            while not self.src_queue.empty():
+                e = self.src_queue.get()
+                if e.terminal and self.retry_pred(e):
+                    if self.task_retries[e.task_id] > 0:
+                        self.run(e.task_config)
+                        self.task_retries = self.task_retries.set(
+                            e.task_id, self.task_retries[e.task_id] - 1)
+                        continue
+                et = e.transform(
+                    ('middleware_data', 'retries'),
+                    self.retries - self.task_retries[e.task_id])
+                self.dest_queue.put(et)
+
+            if self.stopping:
+                return
+
+            time.sleep(1)
+
+    def run(self, task_config):
```

Comment:
In this TaskExecutor's __init__ method:

```
self.TASK_CONFIG_INTERFACE = executor.TASK_CONFIG_INTERFACE
```

TASK_CONFIG_INTERFACE is the way that callers of TaskExecutors know what parameters are needed by run.